### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by tessalt**
fixes a few typos that @michelleyschen noticed, and one that was causing https://github.com/Shopify/buy-button/issues/2097#event-884527839

@michelleyschen @tanema @harisaurus

### CODE REVIEW

File: src/view.js [hunk 1]
```diff
@@ -28,6 +28,10 @@ export default class View {
       name: this.component.name,
       width: this.component.options.layout === 'vertical' ? this.component.options.width : null,
     });
+    this.iframe.el.onload = () => {
+      this.iframe.el.onload = null;
+      this.component.init();
+    };
     this.component.node.className += ` shopify-buy-frame shopify-buy-frame--${this.component.typeKey}`;
```
Comment (michelleyschen) [inline src/view.js]: would it matter that is may not run after `delegateEvents` anymore?

Comment (tessalt) [inline src/view.js]: good point I'll check

**Comment by tessalt** [comment]
fixed up some further issues and did more testing

File: src/views/toggle.js [hunk 2]
```diff
@@ -19,6 +19,10 @@ export default class ToggleView extends View {
     return this.component.options.sticky ? 'is-sticky' : 'is-inline';
   }
 
+  get outerHeight() {
+    return `${this.wrapper.clientHeight}px`;
+  }
+
```
Comment (michelleyschen) [inline src/views/toggle.js]: I think 
```
get typeKey() {		
  return 'toggle';		
}
```
is also missing?

Comment (tessalt) [inline src/views/toggle.js]: no that goes in `components/toggle`

**Comment by tessalt** [comment]
also sneaking in a fix for an IE9 bug relating to my naive addClass implementation

**Comment by michelleyschen** [comment]
mm...I am sorry but I am starting to find this really really difficult to review...

**Comment by tessalt** [comment]
yeah that's because it's fixes for some confusing bugs. I'll add some line comments.

File: src/component.js [hunk 3]
```diff
@@ -103,6 +103,17 @@ export default class Component {
...
+   * get styles for component and any components it contains as determined by manifest.
+   * @return {Object} key-value pairs of CSS styles.
+   */
+  get styles() {
+    return this.options.manifest.filter((component) => this.config[component].styles).reduce((hash, component) => {
+      hash[component] = this.config[component].styles;
+      return hash;
+    }, {});
+  }
+
```
Comment (tessalt) [inline src/component.js]: i shouldn't have moved this in the first place, it works better here.

File: src/utils/element-class.js [hunk 4]
```diff
@@ -5,6 +5,10 @@ export function addClassToElement(className, element) {
   if (element.classList) {
     element.classList.add(className);
   } else {
+    const classes = element.className.split(' ');
+    if (classes.indexOf(className) > -1) {
+      return;
+    }
```
Comment (tessalt) [inline src/utils/element-class.js]: weird event behaviour in IE9 was causing some classes to get added twice, so then not enough of them would get rmeoved. this only adds classes if they don't already exist.

Comment (harisaurus) [inline src/utils/element-class.js]: so weird - i thought classes wouldn't double up. darn you IE

File: src/view.js [hunk 5]
```diff
@@ -64,6 +64,19 @@ export default class View {
         });
       }
     });
+    if (this.iframe) {
+      this.iframe.el.onload = () => {
+        this.iframe.el.onload = null;
+        this.reloadIframe();
+      };
+    }
```
Comment (harisaurus) [inline src/view.js]: is there a place i can read more about why we had to do this?

**Pull request merged**